pypeman/test.py

```diff
@@ -28,88 +30,62 @@ class PypeTestCase(TestCase):
     loop = None
+    wait_subchans = True
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.addCleanup(self.cleanLoop)
 
-    @classmethod
-    def setUpClass(cls):
-        super().setUpClass()
-
-        # Create class event loop used for tests to avoid failing
-        # previous tests to impact next test ? (Not sure)
-        cls.loop = asyncio.new_event_loop()
-
-        # Remove thread event loop to be sure we are not using
-        # another event loop somewhere
-        asyncio.set_event_loop(None)
-
-        # Start channels
-        for chan in channels.all_channels:
-            chan.loop = cls.loop
-            cls.loop.run_until_complete(chan.start())
-            chan._reset_test()
-
-    @classmethod
-    def cleanLoop(cls):
-        """
-        Replace current loop by a new one to avoid side effect on
-        next test.
-        """
-        for chan in channels.all_channels:
-            cls.loop.run_until_complete(chan.stop())
-
-        pending = asyncio.all_tasks(loop=cls.loop)
-        if pending:
-            asyncio.gather(*pending).cancel()
-
-        cls.loop.close()
-        cls.loop = asyncio.new_event_loop()
+    def setUp(self):
+        if self.loop:
+            raise Exception("a loop currently exists at setuP, not normal")
+        self.loop = asyncio.new_event_loop()
+        asyncio.set_event_loop(self.loop)
 
         # Start channels
         for chan in channels.all_channels:
-            chan.loop = cls.loop
-            cls.loop.run_until_complete(chan.start())
+            chan.loop = self.loop
+            chan.wait_subchans = self.wait_subchans
+            self.loop.run_until_complete(chan.start())
             chan._reset_test()
 
-    @classmethod
-    def tearDownClass(cls):
-        super().tearDownClass()
-
-        # Stop channels
+    def cleanLoop(self):
         for chan in channels.all_channels:
-            cls.loop.run_until_complete(chan.stop())
-
-        cls.finish_all_tasks()
+            self.loop.run_until_complete(chan.stop())
+        self.loop.close()
+        self.loop.stop()
+        self.loop = None
+        asyncio.set_event_loop(None)
 
-    @classmethod
-    def finish_all_tasks(cls):
+    def finish_all_tasks(self):
         """
         You can use this function if you have some subchannel in you channel
-        and want to see the final result by processing all remaining tasks.
-        TODO: maybe remove this function (all subchannels are awaited in the
-            handle so pending will be empty)
+        , don't have wait_suchans param set to True and anyway want to see the
+        final result by processing all remaining tasks.
 
         :return: A list of raised exceptions during task execution.
         """
+        if self.wait_subchans:
+            logger.warning(
+                "PypeTestCase.finish_all_tasks called when wait_subchans is set..."
+                "unuseful there will not have pending tasks"
```

Comment:
unuseful -> useless

there will not habe pending tasks

-> there will not be (won't be) any pending tasks  

or 

-> XXX will not have any pending tasks

XXX = channel?